### DISCUSSION TITLE
Category Definition API

### DISCUSSION THREAD

**Discussion by kiranharidas187** [Sunbird-Knowlg/Community#10] [Q&A] [answered]
@pallakartheekreddy We have set up a new sunbird instance for Shikshalokam. When we try to update the Category definition for the course it's giving us some errors. Please let us know what could be the reason

Curl
`curl --location -g --request PATCH '{{base_url}}object/category/definition/v1/update/obj-cat:course_collection_all' \
--header 'Authorization: Bearer {{token}}' \
--header 'Content-Type: application/json' \
--data-raw '{
    "request": {
        "objectCategoryDefinition": {
            "objectMetadata": {
                "config": {
                    "frameworkMetadata": {
                        "orgFWType": [
                            "K-12"
                        ],
                        "targetFWType": [
                            "K-12"
                        ]
                    },
                    "sourcingSettings": {
                        "collection": {
                            "maxDepth": 4,
                            "objectType": "Collection",
                            "primaryCategory": "Course",
                            "isRoot": true,
                            "iconClass": "fa fa-book",
                            "children": {},
                            "hierarchy": {
                                "level1": {
                                    "name": "Course Unit",
                                    "type": "Unit",
                                    "mimeType": "application/vnd.ekstep.content-collection",
                                    "contentType": "CourseUnit",
                                    "primaryCategory": "Course Unit",
                                    "iconClass": "fa fa-folder-o",
                                    "children": {
                                        "Content": []
                                    }
                                },
                                "level2": {
                                    "name": "Course Unit",
                                    "type": "Unit",
                                    "mimeType": "application/vnd.ekstep.content-collection",
                                    "contentType": "CourseUnit",
                                    "primaryCategory": "Course Unit",
                                    "iconClass": "fa fa-folder-o",
                                    "children": {
                                        "Content": []
                                    }
                                },
                                "level3": {
                                    "name": "Course Unit",
                                    "type": "Unit",
                                    "mimeType": "application/vnd.ekstep.content-collection",
                                    "contentType": "CourseUnit",
                                    "primaryCategory": "Course Unit",
                                    "iconClass": "fa fa-folder-o",
                                    "children": {
                                        "Content": []
                                    }
                                },
                                "level4": {
                                    "name": "Course Unit",
                                    "type": "Unit",
                                    "mimeType": "application/vnd.ekstep.content-collection",
                                    "contentType": "CourseUnit",
                                    "primaryCategory": "Course Unit",
                                    "iconClass": "fa fa-folder-o",
                                    "children": {
                                        "Content": []
                                    }
                                }
                            }
                        }
                    }
                },
                "schema": {
                    "properties": {
                        "trackable": {
                            "type": "object",
                            "properties": {
                                "enabled": {
                                    "type": "string",
                                    "enum": [
                                        "Yes",
                                        "No"
                                    ],
                                    "default": "Yes"
                                },
                                "autoBatch": {
                                    "type": "string",
                                    "enum": [
                                        "Yes",
                                        "No"
                                    ],
                                    "default": "Yes"
                                }
                            },
                            "default": {
                                "enabled": "Yes",
                                "autoBatch": "Yes"
                            },
                            "additionalProperties": false
                        },
                        "monitorable": {
                            "type": "array",
                            "items": {
                                "type": "string",
                                "enum": [
                                    "progress-report",
                                    "score-report"
                                ]
                            }
                        },
                        "credentials": {
                            "type": "object",
                            "properties": {
                                "enabled": {
                                    "type": "string",
                                    "enum": [
                                        "Yes",
                                        "No"
                                    ],
                                    "default": "Yes"
                                }
                            },
                            "default": {
                                "enabled": "Yes"
                            },
                            "additionalProperties": false
                        },
                        "userConsent": {
                            "type": "string",
                            "enum": [
                                "Yes",
                                "No"
                            ],
                            "default": "Yes"
                        },
                        "mimeType": {
                            "type": "string",
                            "enum": [
                                "application/vnd.ekstep.content-collection"
                            ]
                        },
                        "generateDIALCodes": {
                            "type": "string",
                            "enum": [
                                "Yes",
                                "No"
                            ],
                            "default": "Yes"
                        },
                        "discussionForum": {
                            "type": "object",
                            "properties": {
                                "enabled": {
                                    "type": "string",
                                    "enum": [
                                        "Yes",
                                        "No"
                                    ],
                                    "default": "Yes"
                                }
                            },
                            "default": {
                                "enabled": "Yes"
                            },
                            "additionalProperties": false
                        }
                    }
                }
            },
            "name": "Course",
            "forms": {
                "childMetadata": {
                    "templateName": "",
                    "required": [],
                    "properties": [
                        {
                            "code": "name",
                            "editable": true,
                            "displayProperty": "Editable",
                            "dataType": "text",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1 required"
                            },
                            "description": "Name",
                            "index": 1,
                            "label": "Name",
                            "required": true,
                            "name": "Name",
                            "inputType": "text",
                            "placeholder": "Name",
                            "validations": [
                                {
                                    "type": "maxLength",
                                    "value": "50",
                                    "message": "Input is Exceeded"
                                },
                                {
                                    "type": "required",
                                    "message": "Name is required"
                                }
                            ]
                        },
                        {
                            "code": "description",
                            "dataType": "text",
                            "description": "Description of the content",
                            "editable": true,
                            "inputType": "textarea",
                            "label": "Description",
                            "name": "Description",
                            "placeholder": "Description",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "required": false,
                            "visible": true
                        },
                        {
                            "code": "primaryCategory",
                            "dataType": "text",
                            "description": "Type",
                            "editable": false,
                            "renderingHints": {},
                            "inputType": "select",
                            "label": "Category",
                            "name": "Type",
                            "placeholder": "",
                            "required": true,
                            "visible": true,
                            "validations": []
                        },
                        {
                            "code": "additionalCategories",
                            "visible": true,
                            "editable": true,
                            "displayProperty": "Editable",
                            "dataType": "list",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "description": "Subject of the Content to use to teach",
                            "index": 7,
                            "label": "Content Additional Categories",
                            "required": false,
                            "name": "additionalCategories",
                            "inputType": "nestedselect",
                            "placeholder": "Content Additional Categories"
                        },
                        {
                            "code": "subjectIds",
                            "visible": true,
                            "depends": [],
                            "editable": false,
                            "dataType": "list",
                            "sourceCategory": "subject",
                            "output": "identifier",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "description": "",
                            "label": "Subjects covered in the course",
                            "required": true,
                            "name": "Subject",
                            "inputType": "nestedselect",
                            "placeholder": "Select Subject"
                        },
                        {
                            "code": "topicsIds",
                            "visible": true,
                            "editable": true,
                            "dataType": "list",
                            "depends": [
                                "subjectIds"
                            ],
                            "sourceCategory": "topic",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "name": "Topic",
                            "description": "Choose a Topics",
                            "inputType": "topicselector",
                            "label": "Topic(s)",
                            "placeholder": "Choose Topics",
                            "required": false,
                            "output": "identifier"
                        },
                        {
                            "code": "copyright",
                            "dataType": "text",
                            "description": "Copyright",
                            "editable": true,
                            "index": 4,
                            "inputType": "text",
                            "label": "Copyright and Year:",
                            "name": "Copyright",
                            "placeholder": "Enter Copyright and Year",
                            "tooltip": "If you are an individual, creating original content, you are the copyright holder. If you are creating this course content on behalf of an organisation, the organisation may be the copyright holder. ",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1 required"
                            },
                            "required": true,
                            "validations": [
                                {
                                    "type": "required",
                                    "message": "Copyright is required"
                                }
                            ]
                        },
                        {
                            "code": "license",
                            "visible": true,
                            "editable": true,
                            "displayProperty": "Editable",
                            "dataType": "text",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1 required"
                            },
                            "description": "Subject of the Content to use to teach",
                            "index": 6,
                            "label": "License:",
                            "required": true,
                            "name": "license",
                            "inputType": "select",
                            "placeholder": "license",
                            "tooltip": "Choose the more appropriate Creative commons license for this Content. ",
                            "validations": [
                                {
                                    "type": "required",
                                    "message": "License is required"
                                }
                            ]
                        },
                        {
                            "code": "author",
                            "dataType": "text",
                            "description": "Author",
                            "editable": true,
                            "index": 5,
                            "inputType": "text",
                            "label": "Author:",
                            "name": "Author",
                            "placeholder": "Author",
                            "tooltip": "Provide name of creator of this content.",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "required": true,
                            "validations": [
                                {
                                    "type": "required",
                                    "message": "Author is required"
                                }
                            ]
                        },
                        {
                            "code": "attributions",
                            "dataType": "list",
                            "description": "Attributions",
                            "editable": true,
                            "index": 3,
                            "inputType": "text",
                            "label": "Attributions :",
                            "name": "attribution",
                            "placeholder": "",
                            "tooltip": "If you have relied on another work to create this content, provide the name of that creator and the source of that work.",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "required": false
                        },
                        {
                            "code": "contentPolicyCheck",
                            "visible": true,
                            "editable": true,
                            "displayProperty": "Editable",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1 required"
                            },
                            "description": "Content Policy check",
                            "index": 7,
                            "labelHtml": "<p class='\''font-italic'\''>I agree that by submitting / publishing this Content, I confirm that this Content complies with prescribed guidelines, including the Terms of Use and Content Policy and that I consent to publish it under the <a class='\''link font-weight-bold'\''  href='\''https://creativecommons.org/licenses'\'' target='\''_blank'\''>Creative Commons Framework in </a> accordance with the  <a class='\''link font-weight-bold'\''  href='\''/terms-of-use.html'\'' target='\''_blank'\''> Content Policy.</a> I have made sure that I do not violate others'\'' copyright or privacy rights.</p>",
                            "required": true,
                            "name": "contentPolicyCheck",
                            "inputType": "checkbox",
                            "placeholder": "Content Policy Check",
                            "validations": [
                                {
                                    "type": "required",
                                    "message": "Content Policy Check is required"
                                }
                            ]
                        }
                    ]
                },
                "create": {
                    "templateName": "",
                    "required": [],
                    "properties": [
                        {
                            "name": "First Section",
                            "fields": [
                                {
                                    "code": "appIcon",
                                    "dataType": "text",
                                    "description": "appIcon of the content",
                                    "editable": true,
                                    "inputType": "appIcon",
                                    "label": "Icon",
                                    "name": "Icon",
                                    "placeholder": "Icon",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "required": true,
                                    "visible": true
                                },
                                {
                                    "code": "name",
                                    "dataType": "text",
                                    "description": "Name of the content",
                                    "editable": true,
                                    "inputType": "text",
                                    "label": "Title",
                                    "name": "Name",
                                    "placeholder": "Title",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "required": true,
                                    "visible": true,
                                    "validations": [
                                        {
                                            "type": "max",
                                            "value": "120",
                                            "message": "Input is Exceeded"
                                        },
                                        {
                                            "type": "required",
                                            "message": "Title is required"
                                        }
                                    ]
                                },
                                {
                                    "code": "description",
                                    "dataType": "text",
                                    "description": "Description of the content",
                                    "editable": true,
                                    "inputType": "textarea",
                                    "label": "Description",
                                    "name": "Description",
                                    "placeholder": "Description",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1"
                                    },
                                    "required": false,
                                    "visible": true,
                                    "validations": [
                                        {
                                            "type": "max",
                                            "value": "256",
                                            "message": "Input is Exceeded"
                                        }
                                    ]
                                },
                                {
                                    "code": "keywords",
                                    "visible": true,
                                    "editable": true,
                                    "dataType": "list",
                                    "name": "Keywords",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "description": "Keywords for the content",
                                    "inputType": "keywords",
                                    "label": "Keywords",
                                    "placeholder": "Enter Keywords",
                                    "required": false,
                                    "validations": []
                                }
                            ]
                        },
                        {
                            "name": "Second Section",
                            "fields": [
                                {
                                    "code": "primaryCategory",
                                    "dataType": "text",
                                    "description": "Type",
                                    "editable": false,
                                    "renderingHints": {},
                                    "inputType": "select",
                                    "label": "Category",
                                    "name": "Type",
                                    "placeholder": "",
                                    "required": true,
                                    "visible": true,
                                    "validations": []
                                },
                                {
                                    "code": "additionalCategories",
                                    "dataType": "list",
                                    "depends": [
                                        "primaryCategory"
                                    ],
                                    "description": "Additonal Category of the Content",
                                    "editable": true,
                                    "inputType": "nestedselect",
                                    "label": "Additional Category",
                                    "name": "Additional Category",
                                    "placeholder": "Select Additional Category",
                                    "renderingHints": {},
                                    "required": false,
                                    "visible": true
                                }
                            ]
                        },
                        {
                            "name": "Organisation Framework Terms",
                            "fields": [
                                {
                                    "code": "framework",
                                    "visible": true,
                                    "editable": true,
                                    "dataType": "text",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "description": "",
                                    "label": "Course Type",
                                    "required": true,
                                    "name": "Framework",
                                    "inputType": "framework",
                                    "placeholder": "Select Course Type",
                                    "output": "identifier",
                                    "validations": [
                                        {
                                            "type": "required",
                                            "message": "Course Type is required"
                                        }
                                    ]
                                },
                                {
                                    "code": "subjectIds",
                                    "visible": true,
                                    "editable": true,
                                    "dataType": "list",
                                    "depends": [
                                        "framework"
                                    ],
                                    "sourceCategory": "subject",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "description": "",
                                    "label": "Subjects covered in the course",
                                    "required": true,
                                    "name": "Subject",
                                    "inputType": "frameworkCategorySelect",
                                    "placeholder": "Select Subject(s)",
                                    "output": "identifier",
                                    "validations": [
                                        {
                                            "type": "required",
                                            "message": "Subjects Taught is required"
                                        }
                                    ]
                                },
                                {
                                    "code": "topicsIds",
                                    "visible": true,
                                    "editable": true,
                                    "dataType": "list",
                                    "depends": [
                                        "framework",
                                        "subjectIds"
                                    ],
                                    "sourceCategory": "topic",
                                    "renderingHints": {},
                                    "name": "Topic",
                                    "description": "Choose a Topics",
                                    "inputType": "topicselector",
                                    "label": "Topics covered in the course",
                                    "placeholder": "Choose Topics",
                                    "required": false,
                                    "output": "identifier"
                                }
                            ]
                        },
                        {
                            "name": "Target Framework Terms",
                            "fields": [
                                {
                                    "code": "audience",
                                    "dataType": "list",
                                    "description": "Audience",
                                    "editable": true,
                                    "inputType": "nestedselect",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1"
                                    },
                                    "label": "Audience Type",
                                    "name": "Audience Type",
                                    "placeholder": "Select Audience Type",
                                    "required": false,
                                    "visible": true,
                                    "range": [
                                        "Student",
                                        "Teacher",
                                        "Parent",
                                        "Administrator"
                                    ]
                                },
                                {
                                    "code": "targetBoardIds",
                                    "visible": true,
                                    "depends": [],
                                    "editable": true,
                                    "dataType": "list",
                                    "sourceCategory": "board",
                                    "output": "identifier",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "description": "Board",
                                    "label": "Board/Syllabus of the audience",
                                    "required": true,
                                    "name": "Board/Syllabus",
                                    "inputType": "select",
                                    "placeholder": "Select Board/Syllabus",
                                    "validations": [
                                        {
                                            "type": "required",
                                            "message": "Board is required"
                                        }
                                    ]
                                },
                                {
                                    "code": "targetMediumIds",
                                    "visible": true,
                                    "depends": [
                                        "targetBoardIds"
                                    ],
                                    "editable": true,
                                    "dataType": "list",
                                    "sourceCategory": "medium",
                                    "output": "identifier",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "description": "",
                                    "label": "Medium(s) of the audience",
                                    "required": true,
                                    "name": "Medium",
                                    "inputType": "nestedselect",
                                    "placeholder": "Select Medium",
                                    "validations": [
                                        {
                                            "type": "required",
                                            "message": "Medium is required"
                                        }
                                    ]
                                },
                                {
                                    "code": "targetGradeLevelIds",
                                    "visible": true,
                                    "depends": [
                                        "targetBoardIds",
                                        "targetMediumIds"
                                    ],
                                    "editable": true,
                                    "dataType": "list",
                                    "sourceCategory": "gradeLevel",
                                    "output": "identifier",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "description": "Class",
                                    "label": "Class(es) of the audience",
                                    "required": true,
                                    "name": "Class",
                                    "inputType": "nestedselect",
                                    "placeholder": "Select Class",
                                    "validations": [
                                        {
                                            "type": "required",
                                            "message": "Class is required"
                                        }
                                    ]
                                },
                                {
                                    "code": "targetSubjectIds",
                                    "visible": true,
                                    "depends": [
                                        "targetBoardIds",
                                        "targetMediumIds",
                                        "targetGradeLevelIds"
                                    ],
                                    "editable": true,
                                    "dataType": "list",
                                    "sourceCategory": "subject",
                                    "output": "identifier",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "description": "",
                                    "label": "Subject(s) of the audience",
                                    "required": true,
                                    "name": "Subject",
                                    "inputType": "nestedselect",
                                    "placeholder": "Select Subject",
                                    "validations": [
                                        {
                                            "type": "required",
                                            "message": "Subject is required"
                                        }
                                    ]
                                }
                            ]
                        },
                        {
                            "name": "Fourth Section",
                            "fields": [
                                {
                                    "code": "author",
                                    "dataType": "text",
                                    "description": "Author of the content",
                                    "editable": true,
                                    "inputType": "text",
                                    "label": "Author",
                                    "name": "Author",
                                    "placeholder": "Author",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1"
                                    },
                                    "required": false,
                                    "visible": true
                                },
                                {
                                    "code": "attributions",
                                    "dataType": "text",
                                    "description": "Attributions",
                                    "editable": true,
                                    "inputType": "text",
                                    "label": "Attributions",
                                    "name": "Attributions",
                                    "placeholder": "Attributions",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1"
                                    },
                                    "required": false,
                                    "visible": true
                                },
                                {
                                    "code": "copyright",
                                    "dataType": "text",
                                    "description": "Copyright",
                                    "editable": true,
                                    "inputType": "text",
                                    "label": "Copyright",
                                    "name": "Copyright & year",
                                    "placeholder": "Copyright",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "required": true,
                                    "visible": true,
                                    "validations": [
                                        {
                                            "type": "required",
                                            "message": "Copyright is required"
                                        }
                                    ]
                                },
                                {
                                    "code": "copyrightYear",
                                    "dataType": "number",
                                    "description": "Year",
                                    "editable": true,
                                    "inputType": "text",
                                    "label": "Copyright Year",
                                    "name": "Copyright Year",
                                    "placeholder": "Copyright Year",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "required": true,
                                    "visible": true,
                                    "validations": [
                                        {
                                            "type": "required",
                                            "message": "Copyright Year is required"
                                        }
                                    ]
                                },
                                {
                                    "code": "license",
                                    "dataType": "text",
                                    "description": "license",
                                    "editable": true,
                                    "inputType": "select",
                                    "label": "License",
                                    "name": "license",
                                    "placeholder": "Select License",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "required": true,
                                    "visible": true,
                                    "defaultValue": "CC BY 4.0",
                                    "validations": [
                                        {
                                            "type": "required",
                                            "message": "License is required"
                                        }
                                    ]
                                }
                            ]
                        }
                    ]
                },
                "delete": {},
                "publish": {},
                "review": {},
                "search": {
                    "templateName": "",
                    "required": [],
                    "properties": [
                        {
                            "code": "primaryCategory",
                            "dataType": "list",
                            "description": "Type",
                            "editable": true,
                            "default": [],
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "inputType": "nestedselect",
                            "label": "Content Type(s)",
                            "name": "Type",
                            "placeholder": "Select ContentType",
                            "required": false,
                            "visible": true
                        },
                        {
                            "code": "board",
                            "visible": true,
                            "depends": [],
                            "editable": true,
                            "dataType": "list",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "description": "Board",
                            "label": "Board",
                            "required": false,
                            "name": "Board",
                            "inputType": "select",
                            "placeholder": "Select Board",
                            "output": "name"
                        },
                        {
                            "code": "medium",
                            "visible": true,
                            "depends": [
                                "board"
                            ],
                            "editable": true,
                            "dataType": "list",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "description": "",
                            "label": "Medium(s)",
                            "required": false,
                            "name": "Medium",
                            "inputType": "nestedselect",
                            "placeholder": "Select Medium",
                            "output": "name"
                        },
                        {
                            "code": "gradeLevel",
                            "visible": true,
                            "depends": [
                                "board",
                                "medium"
                            ],
                            "editable": true,
                            "default": "",
                            "dataType": "list",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "description": "Class",
                            "label": "Class(es)",
                            "required": false,
                            "name": "Class",
                            "inputType": "nestedselect",
                            "placeholder": "Select Class",
                            "output": "name"
                        },
                        {
                            "code": "subject",
                            "visible": true,
                            "depends": [
                                "board",
                                "medium",
                                "gradeLevel"
                            ],
                            "editable": true,
                            "default": "",
                            "dataType": "list",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "description": "",
                            "label": "Subject(s)",
                            "required": false,
                            "name": "Subject",
                            "inputType": "nestedselect",
                            "placeholder": "Select Subject",
                            "output": "name"
                        },
                        {
                            "code": "topic",
                            "visible": true,
                            "editable": true,
                            "dataType": "list",
                            "depends": [
                                "board",
                                "medium",
                                "gradeLevel",
                                "subject"
                            ],
                            "default": "",
                            "renderingHints": {
                                "class": "sb-g-col-lg-1"
                            },
                            "name": "Topic",
                            "description": "Choose a Topics",
                            "inputType": "topicselector",
                            "label": "Topic(s)",
                            "placeholder": "Choose Topics",
                            "required": false
                        }
                    ]
                },
                "unitMetadata": {
                    "templateName": "",
                    "required": [],
                    "properties": [
                        {
                            "name": "First Section",
                            "fields": [
                                {
                                    "code": "name",
                                    "dataType": "text",
                                    "description": "Name of the content",
                                    "editable": true,
                                    "inputType": "text",
                                    "label": "Title",
                                    "name": "Title",
                                    "placeholder": "Title",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1 required"
                                    },
                                    "required": true,
                                    "visible": true,
                                    "validations": [
                                        {
                                            "type": "max",
                                            "value": "120",
                                            "message": "Input is Exceeded"
                                        },
                                        {
                                            "type": "required",
                                            "message": "Title is required"
                                        }
                                    ]
                                },
                                {
                                    "code": "description",
                                    "dataType": "text",
                                    "description": "Description of the content",
                                    "editable": true,
                                    "inputType": "textarea",
                                    "label": "Description",
                                    "name": "Description",
                                    "placeholder": "Description",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1"
                                    },
                                    "required": false,
                                    "visible": true,
                                    "validations": [
                                        {
                                            "type": "max",
                                            "value": "256",
                                            "message": "Input is Exceeded"
                                        }
                                    ]
                                },
                                {
                                    "code": "keywords",
                                    "visible": true,
                                    "editable": true,
                                    "dataType": "list",
                                    "name": "Keywords",
                                    "renderingHints": {
                                        "class": "sb-g-col-lg-1"
                                    },
                                    "index": 3,
                                    "description": "Keywords for the content",
                                    "inputType": "keywords",
                                    "label": "Keywords",
                                    "placeholder": "Enter Keywords",
                                    "required": false,
                                    "validations": []
                                },
                                {
                                    "code": "topic",
                                    "visible": true,
                                    "depends": [],
                                    "editable": true,
                                    "dataType": "list",
                                    "renderingHints": {},
                                    "name": "Topic",
                                    "description": "Choose a Topics",
                                    "index": 11,
                                    "inputType": "topicselector",
                                    "label": "Topics",
                                    "placeholder": "Choose Topics",
                                    "required": false,
                                    "validations": []
                                }
                            ]
                        }
                    ]
                },
                "update": {
                    "templateName": "",
                    "required": [],
                    "properties": [
                        {
                            "code": "name",
                            "editable": true,
                            "displayProperty": "Editable",
                            "dataType": "text",
                            "renderingHints": {
                                "semanticColumnWidth": "twelve"
                            },
                            "description": "Name",
                            "index": 1,
                            "label": "Name",
                            "required": false,
                            "name": "Name",
                            "inputType": "text",
                            "placeholder": "Name"
                        },
                        {
                            "code": "board",
                            "depends": [
                                "medium",
                                "gradeLevel",
                                "subject"
                            ],
                            "editable": true,
                            "displayProperty": "Editable",
                            "dataType": "text",
                            "renderingHints": {
                                "semanticColumnWidth": "six"
                            },
                            "description": "Education Board (Like MP Board, NCERT, etc)",
                            "index": 2,
                            "label": "Board",
                            "required": false,
                            "name": "Board",
                            "inputType": "select",
                            "placeholder": "Board"
                        },
                        {
                            "code": "medium",
                            "depends": [
                                "gradeLevel",
                                "subject"
                            ],
                            "editable": true,
                            "dataType": "list",
                            "renderingHints": {
                                "semanticColumnWidth": "six"
                            },
                            "description": "Medium of instruction",
                            "index": 3,
                            "label": "Medium",
                            "required": false,
                            "name": "Medium",
                            "inputType": "multiSelect",
                            "placeholder": "Medium"
                        },
                        {
                            "code": "gradeLevel",
                            "depends": [
                                "subject"
                            ],
                            "editable": true,
                            "dataType": "list",
                            "renderingHints": {
                                "semanticColumnWidth": "six"
                            },
                            "description": "Grade",
                            "index": 4,
                            "label": "Grade",
                            "required": false,
                            "name": "Grade",
                            "inputType": "multiSelect",
                            "placeholder": "Grade"
                        },
                        {
                            "code": "subject",
                            "editable": true,
                            "dataType": "list",
                            "renderingHints": {
                                "semanticColumnWidth": "six"
                            },
                            "name": "Subject",
                            "description": "Subject of the Content to use to teach",
                            "index": 5,
                            "inputType": "multiSelect",
                            "label": "Subject",
                            "placeholder": "Grade",
                            "required": false
                        },
                        {
                            "code": "learningOutcome",
                            "dataType": "list",
                            "description": "",
                            "editable": true,
                            "index": 6,
                            "inputType": "select",
                            "label": "Learning Outcome :",
                            "name": "Learning Outcome :",
                            "placeholder": "Select Learning Outcome",
                            "depends": [
                                "topic"
                            ],
                            "renderingHints": {},
                            "required": false
                        },
                        {
                            "code": "attributions",
                            "dataType": "list",
                            "description": "Attributions",
                            "editable": true,
                            "index": 7,
                            "inputType": "text",
                            "label": "Attributions",
                            "name": "attribution",
                            "placeholder": "",
                            "tooltip": "If you have relied on another work to create this content, provide the name of that creator and the source of that work.",
                            "renderingHints": {},
                            "required": false
                        },
                        {
                            "code": "author",
                            "dataType": "text",
                            "description": "Author",
                            "editable": true,
                            "index": 8,
                            "inputType": "text",
                            "label": "Author",
                            "name": "Author",
                            "placeholder": "Author",
                            "tooltip": "Provide name of creator of this content.",
                            "renderingHints": {},
                            "required": false
                        },
                        {
                            "code": "copyright",
                            "dataType": "text",
                            "description": "Copyright",
                            "editable": true,
                            "index": 9,
                            "inputType": "text",
                            "label": "Copyright",
                            "name": "Copyright",
                            "placeholder": "Copyright",
                            "tooltip": "If you are an individual, creating original content, you are the copyright holder. If you are creating this course content on behalf of an organisation, the organisation may be the copyright holder. ",
                            "renderingHints": {},
                            "required": false
                        }
                    ]
                }
            }
        }
    }
}'`

Response

`{
    "id": "api.object.category.definition.update",
    "ver": "3.0",
    "ts": "2022-05-17T13:22:48ZZ",
    "params": {
        "resmsgid": "59cca1c2-b5a6-4be1-944d-d2b7247beb48",
        "msgid": null,
        "err": "CLIENT_ERROR",
        "status": "failed",
        "errmsg": "Validation Errors"
    },
    "responseCode": "CLIENT_ERROR",
    "result": {
        "messages": [
            "Metadata visibility cannot have new property with name visibility"
        ]
    }
}`

**Comment by Krishnaj20** [comment]
@AmiableAnil @Jayaprakash8887 - pls look into this open query.

**Comment by pallakartheekreddy** [comment, marked as answer]
@kiranharidas187 Please check if the `objectcategorydefinition` schema is up to date in the environment?
As per the above error `visibility` property is not available in `objectcategorydefinition` schema.
@Krishnaj20

**Reply by Krishnaj20** [reply]
@kiranharidas187 - Hope your question is clarified?

**Reply by kiranharidas187** [reply]
yes @Krishnaj20. The schema was not updated in our case. After updating it started working. Thanks @pallakartheekreddy

**Reply by pramodkvarma** [reply]
@kiranharidas187 See if you can do an update to the doc and do a PR. Good way to help open source projects!

**Reply by kiranharidas187** [reply]
sure @pramodkvarma.  Can I get some doc on know the repo details and set up details?

**Reply by pramodkvarma** [reply]
@Krishnaj20 can you help point to the docs for these topics? Ideally we should continuosly get docs fixed by community.

**Reply by Jayaprakash8887** [reply]
https://github.com/project-sunbird/knowledge-platform/tree/master/schemas/content/1.0

https://knowlg.sunbird.org/ --> 'Product & Developer Guide' --> 'Content Service' --> Schemas
https://knowlg.sunbird.org/ --> 'Installation guide' --> Services
